src/main/java/org/jvnet/hudson/test/SimpleCommandLauncher.java

```diff
@@ -72,29 +73,37 @@ public void launch(SlaveComputer computer, final TaskListener listener) {
             }
             listener.getLogger().println("$ " + cmd);
             ProcessBuilder pb = new ProcessBuilder(Util.tokenize(cmd));
-            final EnvVars cookie = EnvVars.createCookie();
+            cookie = EnvVars.createCookie();
             pb.environment().putAll(cookie);
             if (env != null) {
             	pb.environment().putAll(env);
             }
-            final Process proc = pb.start();
+            proc = pb.start();
             new StreamCopyThread("stderr copier for remote agent on " + computer.getDisplayName(), proc.getErrorStream(), listener.getLogger()).start();
-            computer.setChannel(proc.getInputStream(), proc.getOutputStream(), listener.getLogger(), new Channel.Listener() {
-                @Override
-                public void onClosed(Channel channel, IOException cause) {
-                    try {
-                        ProcessTree.get().killAll(proc, cookie);
-                    } catch (Exception x) {
-                        LOGGER.log(Level.WARNING, null, x);
-                    }
-                }
-            });
+            computer.setChannel(proc.getInputStream(), proc.getOutputStream(), listener, null);
             LOGGER.log(Level.INFO, "agent launched for {0}", computer.getName());
         } catch (Exception x) {
             LOGGER.log(Level.WARNING, null, x);
         }
     }
 
+    @SuppressFBWarnings(value = "IS2_INCONSISTENT_SYNC", justification = "test code, close enough")
```

Comment:
`command-launcher` plugin could be updated to match, I guess, though it mainly matters here because of Windows tests.